core/dbt/config/runtime.py

```diff
@@ -109,17 +73,18 @@ def from_parts(cls, project, profile, args):
             config=profile.config,
             threads=profile.threads,
             credentials=profile.credentials,
-            args=args
+            args=args,
+            cli_vars=cli_vars,
```

Comment:
What's up with this line? Were we not supplying the cli_vars before? Or is this also to appease mypy?

Reply:
Because they're all dataclasses the actual class _does_ now take a `cli_vars` argument, and it isn't optional. Now it's the responsibility of classmethods instead of `__init__`. Before we set `cli_vars` in `__init__`.